### PR TITLE
Stateful norms fix up

### PR DESCRIPTION
**Pull request by Aarsh2001**

### CODE REVIEW

File: ivy_tests/test_ivy/test_stateful/test_norms.py [hunk 1]
```diff
@@ -1,56 +1,64 @@
...
-            [[-1.2247356, 0.0, 1.2247356], [-1.2247356, 0.0, 1.2247356]],
-        ),
-    ],
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float")),
+    new_std=st.floats(min_value=0.0, max_value=1.0),
+    init_with_v=st.booleans(),
+    method_with_v=st.booleans(),
+    num_positional_args_init=helpers.num_positional_args(fn_name="LayerNorm.__init__"),
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_stateful/test_norms.py:21]: I think we don't need to use this helper as we have pre-defined positional-only and keyword-only arguments, please pass the number of positional-only args to the test_method as a constant directly.

File: ivy_tests/test_ivy/test_stateful/test_norms.py [hunk 2]
```diff
@@ -1,56 +1,64 @@
...
-        return
-    assert np.allclose(ivy.to_numpy(norm_layer(x)), ivy.to_numpy(target))
-    # compilation test
-    if ivy.current_backend_str() == "torch":
-        # this is not a backend implemented function
-        return
+    input_dtype, x = dtype_and_x
+    x = np.asarray(x, dtype=input_dtype)
+    shape = x.shape
+    print(f"\nbefore -: {input_dtype}")
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_stateful/test_norms.py:43]: debug statement

File: ivy_tests/test_ivy/test_stateful/test_optimizers.py [hunk 3]
```diff
@@ -176,10 +176,6 @@ def test_adam_optimizer(
         container_flags_method=container,
         method_with_v=with_v,
         device_=device,
-        all_as_kwargs_np_method={
```
Comment (VedPatwardhan) [inline ivy_tests/test_ivy/test_stateful/test_optimizers.py:179]: Please revert this change as it's related to a different submodule

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @Aarsh2001, I have suggested a few changes, please take a look 🙂

**Review by VedPatwardhan — APPROVED** [review]
Hi @Aarsh2001, the changes look good to me, I'll just request Dan's review on this PR now 🙂

**Review by djl11 — APPROVED** [review]
lgtm!

**Pull request merged**